x-pack/plugins/security_solution/public/management/components/endpoint_execute_action/execute_action_host_response_output.tsx

```diff
@@ -0,0 +1,142 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { memo, useMemo } from 'react';
+import {
+  EuiAccordion,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiText,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ResponseActionFileDownloadLink } from '../response_action_file_download_link';
+import type {
+  ActionDetails,
+  MaybeImmutable,
+  ResponseActionExecuteOutputContent,
+} from '../../../../common/endpoint/types';
+
+const EXECUTE_FILE_LINK_TITLE = i18n.translate(
+  'xpack.securitySolution.responseActionExecuteDownloadLink.downloadButtonLabel',
+  { defaultMessage: 'Click here to download full output' }
+);
+
+const ACCORDION_BUTTON_TEXT = Object.freeze({
+  output: {
+    regular: i18n.translate(
+      'xpack.securitySolution.responseActionExecuteAccordion.outputButtonTextRegular',
+      {
+        defaultMessage: 'Execution output',
+      }
+    ),
+    truncated: i18n.translate(
+      'xpack.securitySolution.responseActionExecuteAccordion.outputButtonTextTruncated',
+      {
+        defaultMessage: 'Execution output (truncated)',
+      }
+    ),
+  },
+  error: {
+    regular: i18n.translate(
+      'xpack.securitySolution.responseActionExecuteAccordion.errorButtonTextRegular',
+      {
+        defaultMessage: 'Execution error',
+      }
+    ),
+    truncated: i18n.translate(
+      'xpack.securitySolution.responseActionExecuteAccordion.errorButtonTextTruncated',
+      {
+        defaultMessage: 'Execution error (truncated)',
+      }
+    ),
+  },
+});
+interface ExecuteActionOutputProps {
+  content: string;
+  initialIsOpen?: boolean;
+  isTruncated: boolean;
+  type: 'error' | 'output';
+}
+
+const ExecutionActionOutputAccordion = memo<ExecuteActionOutputProps>(
+  ({ content, initialIsOpen = false, isTruncated, type }) => {
+    const id = useGeneratedHtmlId({
+      prefix: 'executeActionOutputAccordions',
+      suffix: type,
+    });
+    return (
+      <EuiAccordion
+        id={id}
+        initialIsOpen={initialIsOpen}
+        buttonContent={ACCORDION_BUTTON_TEXT[type][isTruncated ? 'truncated' : 'regular']}
+        paddingSize="s"
+      >
+        <EuiText
+          size="s"
+          style={{
+            whiteSpace: 'pre-wrap',
+            lineBreak: 'anywhere',
+          }}
+        >
+          <p>{content}</p>
+        </EuiText>
+      </EuiAccordion>
+    );
+  }
+);
+ExecutionActionOutputAccordion.displayName = 'ExecutionActionOutputAccordion';
+
+interface ExecuteActionHostResponseOutputProps {
+  action: MaybeImmutable<ActionDetails>;
+  agentId?: string;
+  'data-test-subj'?: string;
+  textSize?: 's' | 'xs';
+}
+
+export const ExecuteActionHostResponseOutput = memo<ExecuteActionHostResponseOutputProps>(
+  ({ action, agentId, 'data-test-subj': dataTestSubj, textSize }) => {
+    const outputContent = useMemo(
```

Comment:
`agentId` is optional. can you add code here to default it to the first agent id that the action was sent to case this component is used without setting an `agentId`?

Reply:
done fdac60dd7e5fc6cb3b807d4b10534da927e12fb3